airflow/configuration.py

```diff
@@ -580,10 +589,24 @@ def getint(self, section, key):
     def getfloat(self, section, key):
         return float(self.get(section, key))
 
+    def remove_option(self, section, option):
+        try:
+            ConfigParser.remove_option(self, section, option)
+        except ConfigParser.NoSectionError:
+            pass
+        try:
+            ConfigParser.remove_option(self.defaults, section, option)
+        except ConfigParser.NoSectionError:
+            pass
+
     def read(self, filenames):
         ConfigParser.read(self, filenames)
         self._validate()
 
+    def has_option(self, section, option):
```

Comment:
Yes, I completely agree, the thing we need to determine is the behavior of `has_option`. Should `has_option` return True for `'something'` if `'something_cmd'` is defined?

If yes, then there will be no way for people to check if `'something_cmd'` is defined but `'something'` is not.

If `has_option` returns true for values defined on `self` and on `self.defaults` the only way to check if there's a default but no explicit value will be to check: `self.get('X') == self.defaults.get('X')`.

If we leave `has_option` without overriding, we will get some false negatives but we will not get any false positives.